### PR TITLE
[libc++] mark P0154 as implemented; fix column alignment issues

### PR DESCRIPTION
**Pull request by h-vetinari**
AFAIU, P0154 support is complete as of https://github.com/llvm/llvm-project/pull/89446

### CODE REVIEW

File: libcxx/docs/Status/Cxx17.rst
Comment (mordante) [inline libcxx/docs/Status/Cxx17.rst:45]: When LLVM-19 is released libc++ still supports Clang 18, which does not have the needed support. Therefore I prefer to keep the note and mention it requires Clang 19 or newer.

(I've no objection to mark the feature as complete.)

**Review by mordante — APPROVED** [review]
Thanks for catching this! LGTM modulo one comment.

File: libcxx/docs/Status/Cxx17.rst
Comment (h-vetinari) [reply libcxx/docs/Status/Cxx17.rst:45]: Done

**Comment by mordante** [comment]
Do you have commit access?

**Comment by h-vetinari** [comment]
> Do you have commit access?

I do not... 🙃

File: libcxx/docs/Status/Cxx17Papers.csv
Comment (h-vetinari) [inline libcxx/docs/Status/Cxx17Papers.csv:38]: Now that we've reinstated the footnote, we _may_ want to keep the 15.0 as the version when this was implemented? 🤔

Comment (mordante) [reply libcxx/docs/Status/Cxx17Papers.csv:38]: I prefer to keep 19, 15 only works with GCC. The combination of GCC and libc++ is not too common.

**Pull request merged**
